packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -305,54 +311,68 @@ export function importAnalysisPlugin(config: ResolvedConfig): Plugin {
           )
         }
 
-        const isRelative = url.startsWith('.')
-        const isSelfImport = !isRelative && cleanUrl(url) === cleanUrl(importer)
+        let resolvedId = resolved.id
+        if (isExternalUrl(resolvedId)) {
+          return [resolvedId, resolvedId, resolved.meta]
+        }
+
+        const isRelativeUrl = /^\.\.?(?:$|\/)/.test(url)
+        const isSelfImport =
+          !isRelativeUrl && cleanUrl(url) === cleanUrl(importer)
+        const isPreBundled = resolvedId.startsWith(
+          getDepsCacheDirPrefix(config)
+        )
+        const isFileInsideRoot =
+          !isPreBundled && resolvedId.startsWith(root + '/')
+        const isFileOutsideRoot =
+          !isPreBundled &&
+          !isFileInsideRoot &&
+          fs.existsSync(cleanUrl(resolvedId))
 
         // normalize all imports into resolved URLs
         // e.g. `import 'foo'` -> `import '/@fs/.../node_modules/foo/index.js'`
-        if (resolved.id.startsWith(root + '/')) {
+        let resolvedUrl: string
+        if (isPreBundled || isFileOutsideRoot) {
```

Comment:
I swapped the conditions so that `isPreBundled || isFileOutsideRoot` is checked before `isFileInsideRoot`, since it *was* possible (even common!) for an optimized dependency file to exist within the project root.